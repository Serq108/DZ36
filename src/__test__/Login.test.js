import React from 'react';
import {shallow, mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import '../../jest.config'
import {Login} from '../containers/login/Login'
import {MemoryRouter} from "react-router-dom";
import Service from '../service/service';


describe('Login', () => {
    let output;

    beforeEach(() => {
        output = mount(
            <MemoryRouter>
                <Login/>
            </MemoryRouter>
        ).find('Login');
        fetch.resetMocks();
    });

    it('should render login container', () => {
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    it('should change state on fields change', async () => {
        const mockChangeUsername = {
            target: {
                name: 'username',
                value: 'Ilon'
            }
        };

        const mockChangePassword = {
            target: {
                name: 'password',
                value: '123'
            }
        };

        const expected = {
            username: 'Ilon',
            password: '123',
        };
        
        fetch.mockResponseOnce(JSON.stringify({ data: '12345' }));
        output.instance().handleChangeLog(mockChangeUsername);
        output.instance().handleChangePass(mockChangePassword);
        //await output.instance().submitForm();

        expect(output.state()).toEqual(expected);
    });

    it('fetches data from server when server returns a successful response', async () => {
        fetch.mockResponseOnce(JSON.stringify({ data: '12345' }));
        //assert on the response
        //const data = 'username=mihan&password=123';

        const resp = await Service.getReq('acc/');
        expect(resp.data).toEqual('12345')
        //assert on the times called and arguments given to fetch
        expect(fetch.mock.calls.length).toEqual(1);
    });

    it('login', async () => {
        fetch.mockResponseOnce(JSON.stringify({ data: '12345' }));
        const mockChangeUsername = {
            target: {
                name: 'username',
                value: 'Ilon'
            }
        };

        const mockChangePassword = {
            target: {
                name: 'password',
                value: '123'
            }
        };
        //assert on the response
        const data = 'username=mihan&password=123';

        output.instance().handleChangeLog(mockChangeUsername);
        output.instance().handleChangePass(mockChangePassword);
        //await output.instance().submitForm();
        await Service.authPost(data);
        fetch.mockResponseOnce(JSON.stringify({ data: '12345' }));
        const resp = await Service.getReq('acc/');
        expect(resp.data).toEqual('12345')
        fetch.mockResponseOnce(JSON.stringify({ data: '12345' }));
        await output.instance().handleAcc();    
    });


});
