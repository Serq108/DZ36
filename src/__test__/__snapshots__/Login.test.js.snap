// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render login container 1`] = `
<Login>
  <div
    className="login-form"
  >
    <h1>
      Please Login 
    </h1>
    <Link
      className="reg-link"
      to="/reg"
    >
      <a
        className="reg-link"
        href="/reg"
        onClick={[Function]}
      >
        or register
      </a>
    </Link>
    <input
      onChange={[Function]}
      placeholder="Login"
      type="text"
    />
    <input
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
    />
    <input
      onClick={[Function]}
      type="submit"
      value="log in"
    />
  </div>
</Login>
`;
